tessellation/src/basic_shapes.rs

```diff
@@ -715,39 +715,49 @@ pub fn stroke_circle<Output>(center: Point, radius: f32, tolerance: f32, output:
 }
 
 /// Tessellate a convex polyline.
-///
-/// TODO: normals are not implemented yet.
-pub fn fill_convex_polyline<Iter, Output>(mut it: Iter, output: &mut Output) -> Count
+// We insert 2nd point on line first in order to have the neighbours for normal calculation.
+pub fn fill_convex_polyline<'a, Iter, Output>(it: &'a Iter, output: &mut Output) -> Count
 where
-    Iter: Iterator<Item = Point>,
+    &'a Iter: IntoIterator<Item = &'a Point>,
+    <&'a Iter as IntoIterator>::IntoIter: Clone,
```

Comment:
More generally, what motivated the change from taking an iterator to taking a &IntoIterator?